pkg/tls/tls.go

```diff
@@ -81,27 +88,71 @@ func NewCert(
 		SerialNumber:          serialNumber,
 		NotBefore:             now,
 		NotAfter:              now.AddDate(2, 0, 0),
-		Subject:               pkix.Name{CommonName: cn, Organization: organization},
+		Subject:               p.Subject,
 		BasicConstraintsValid: true,
 		KeyUsage:              x509.KeyUsageDigitalSignature | x509.KeyUsageKeyEncipherment,
-		ExtKeyUsage:           extKeyUsage,
-		DNSNames:              dnsNames,
-		IPAddresses:           ipAddresses,
+		ExtKeyUsage:           p.ExtKeyUsage,
+		DNSNames:              p.DNSNames,
+		IPAddresses:           p.IPAddresses,
 	}
 
 	key, err := NewPrivateKey()
 	if err != nil {
 		return nil, nil, err
 	}
-	var cert *x509.Certificate
-	if selfSign {
-		cert, err = newCert(key, template, nil, nil)
-	} else {
-		cert, err = newCert(key, template, signingkey, signingcert)
-	}
+	cert, err := newCert(key, template, p.SigningKey, p.SigningCert)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	return key, cert, nil
 }
+
+// CertMatchesParams returns true if the given key and cert match the CertParams
+// struct.
+func CertMatchesParams(key *rsa.PrivateKey, cert *x509.Certificate, params *CertParams) bool {
+	if key == nil || cert == nil {
+		return false
+	}
+
+	// check the key and cert match eachother
+	certPublicKey, ok := cert.PublicKey.(*rsa.PublicKey)
+	if !ok || !reflect.DeepEqual(key.PublicKey, *certPublicKey) {
```

Comment:
Is it possible to get a different public key when the only variable input to our certgen process is the dns names? IOW, isn't it enough to compare the dns names only?

Reply:
I guess we also need to account for different signing keys.

Reply:
it's belt and braces